test/puppetlabs/puppetdb/testutils.clj

```diff
@@ -219,54 +219,74 @@
               :body body
               :params params))))
 
+(defn query-request
+  "Create a ring request map for a PuppetDB query. `http-method` indicates :get or :post.
+  Parameters or headers are optional"
+  ([http-method path query]
+   (query-request http-method path query {}))
+  ([http-method path query {:keys [params headers]}]
+   (if (= :get http-method)
+     (get-request path query params headers)
+     (post-request path nil nil headers
+                   (-> params
+                       (assoc :query query)
+                       json/generate-string
+                       (.getBytes "UTF-8")
+                       java.io.ByteArrayInputStream.)))))
+
 (defn content-type
   "Returns the content type of the ring response"
   [resp]
   (get-in resp [:headers "Content-Type"]))
 
 (defn paged-results*
-  "Makes a ring request to `path` using the `app-fn` ring handler. Sets the necessary parameters
+   "Makes a ring request to `path` using the `app-fn` ring handler. Sets the necessary parameters
    for paged results.  Returns the ring response, with the body converted from the stream/JSON
-   to clojure data structures."
-  [{:keys [app-fn path query params limit total include_total offset] :as paged-test-params}]
-  {:pre [(= #{} (difference
-                 (keyset paged-test-params)
-                 #{:app-fn :path :query :params :limit :total :include_total :offset}))]}
-  (let [params  (merge params
-                       {:limit limit
-                        :offset offset})
-        request (get-request path query
-                             (if include_total
-                               (assoc params :include_total true)
-                               params))
-        resp (app-fn request)
-        body    (if (string? (:body resp))
-                  (:body resp)
-                  (slurp (:body resp)))]
-    (assoc resp :body (json/parse-string body true))))
+  to clojure data structures."
+   ([paged-test-params]
+    (paged-results* :get paged-test-params))
+   ([method {:keys [app-fn path query params limit total include_total offset] :as paged-test-params}]
+    {:pre [(= #{} (difference
+                   (keyset paged-test-params)
+                   #{:app-fn :path :query :params :limit :total :include_total :offset}))]}
+    (let [params  (merge params
+                         {:limit limit
+                          :offset offset})
+          request (query-request method path query
```

Comment:
Here, the comment on github is helpful. Adding the arity overload is a good change, but does clearly make the diff messy from github's review UI; the comment helps bme figure out where to look.